### DISCUSSION TITLE
Testing components with @Attribute

### DISCUSSION THREAD

**Discussion by maxs-rose** [help-me-mom/ng-mocks#2548] [Q&A] [answered]
Is it possible to test a component with an `@Attribute`?

Assuming we have
```ts
// Component
class Example {
    constructor(@Attribute("test") public readonly test: string) {}
}
```

I would assume that this would be set the same as any other `@Input` attribute but this doesn't seem to work and instead, the attribute is actually null.
```ts
// Tests
beforeEach(() => MockBuilder(Example, Module));

it('test', () => expect(MockRender(Example, { test: "test" }).point.componentInstance.test).toEqual("test");)
```

**Comment by maxs-rose** [comment]
Relevant angular docs https://angular.io/api/core/Attribute

**Comment by satanTime** [comment, marked as answer]
Hi,

as I see, `@Attribute` is static, it means `<input type="text">` injects `text`, whereas `<input type="{{ type }}">` or `<input [type]="type">` inject `null`.

I would suggest to pass a template instead of the class:

```ts
it('test', () => {
  const { point } = MockRender<Example>('<example test="test"></example>');
  expect(point.componentInstance.test).toEqual('test');
});
```

**Reply by maxs-rose** [reply]
Yea that works thanks :)
